### DISCUSSION TITLE
Can I import a newer observableJS plot library?

### DISCUSSION THREAD

**Discussion by libjohn** [quarto-dev/quarto-cli#6410] [Q&A] [answered]
### Description

I have some development Quarto/OJS code that I draft-published to [netlify](https://radiant-dusk-9695f6.netlify.app/).  My problem is with a bit of code that throws errors when render in Quarto, even as it works fine in an Observable notebook.   

Is there a ojs import statement that can get the latest version of the Plot package?  Or any suggestions on how to handle this?

The code in question (below) is commented out so it can be served on Netlify without the hover-labels I want.  The technique comes from this documentation ([docsa](https://observablehq.com/plot/marks/tip), [docsb](https://observablehq.com/plot/interactions/pointer)).  The error messages are:

- TypeError: Plot.pointer is not a function
- TypeError: Plot.tip is not a function

```
    Plot.tip(
      my_filtered_weather,
      Plot.pointer({
        x: "startTime",
        y: "value",
        stroke: "measurement",
        channels: { measurement: "measurement"},
      })
    ),
```

**Comment by mcanouil** [comment]
I don't believe there is a way as it could break Observable integration in Quarto.
I would suggest to use `Plot.version` to see what Quarto is using.

**Comment by libjohn** [comment]
- Plot version is "0.6.0"
    - Not sure if the GitHub releases of Plot are the same numbering, but that is [v0.6.9](https://github.com/observablehq/plot/releases)
- quarto --version is 1.3.433

**Reply by libjohn** [reply, marked as answer]
To update,  It is possible to load the latest Plot library

```
Plot = import("https://esm.run/@observablehq/plot")
```

**Reply by mcanouil** [reply]
I did not know it was possible to import only some parts of Observable.
Thanks for sharing!

**Reply by jimjam-slam** [reply]
Also discussed here! https://github.com/quarto-dev/quarto-cli/discussions/3677#discussioncomment-4422363

**Reply by cscheid** [reply]
Incidentally, thank you for using `esm.run` (for everyone else, that's the latest competitor in npm-to-esm autoconverting CDNs esm.sh, skypack.dev, etc.) Since both esm.sh and skypack.dev have been flaky in the past, it's good to know of other options!

**Comment by mcanouil** [comment]
FYI, I made a PR to document this:

- https://github.com/quarto-dev/quarto-web/pull/802
